psp/ml/models/recent_history.py

```diff
@@ -0,0 +1,190 @@
+import dataclasses
+from collections import defaultdict
+from datetime import datetime, timedelta
+from typing import Iterator, Tuple
+
+import numpy as np
+import xarray as xr
+
+from psp.ml.dataset import PvDataSource
+from psp.ml.models.base import PvSiteModel, PvSiteModelConfig
+from psp.ml.models.regressors.base import Regressor
+from psp.ml.typings import Batch, Features, X, Y
+from psp.pv import get_irradiance
+from psp.utils.maths import safe_div
+
+
+def to_midnight(ts: datetime) -> datetime:
+    return ts.replace(hour=0, minute=0, second=0, microsecond=0)
+
+
+def tod_as_minutes(ts: datetime) -> float:
+    """Time of day as minutes since midnight."""
+    midnight = to_midnight(ts)
+    return (ts - midnight).total_seconds() / 60.0
+
+
+AGGS = ["nanmean", "nanmedian", "nanstd", "nanmin", "nanmax"]
+
+
+@dataclasses.dataclass
+class SetupConfig:
+    data_source: PvDataSource
+
+
+class RecentHistoryModel(PvSiteModel):
+    def __init__(
+        self,
+        config: PvSiteModelConfig,
+        setup_config: SetupConfig,
+        regressor: Regressor,
+    ):
+        self._data_source: PvDataSource
+        self.setup(setup_config)
+        self._regressor = regressor
+        super().__init__(config, setup_config)
+
+    def predict_from_features(self, features: Features) -> Y:
+        pred = self._regressor.predict(features)
+        y = Y(powers=pred * features["factor"] * features["irradiance"])
+        return y
+
+    def _get_tod_stats(
+        self,
+        x: X,
+        yesterday_midnight: datetime,
+        future_interval: Tuple[float, float],
+        data: xr.DataArray,
+    ):
+
```

Comment:
This is probably a general comment, but can we add function descriptions? I think I would define this as necessary for production. Doesnt have to be in this PR, could ust chunk in a issue to do this